### PR TITLE
Updates and fixes to viewport placement

### PR DESCRIPTION
**Pull request by hultonha**
This fix replaces legacy calls to `GetHitLocation` with `CalculateWorldPosition` which uses the new `IntersectorBus`.

The PR also removes legacy viewport calls that are no longer used.

### CODE REVIEW

File: Code/Editor/2DViewport.cpp [hunk 1]
```diff
@@ -234,7 +234,7 @@ void Q2DViewport::UpdateContent(int flags)
 }
 
 //////////////////////////////////////////////////////////////////////////
```
Comment (NicholasVanSickle) [inline Code/Editor/2DViewport.cpp:236]: I think we can redcode this class btw

File: Code/Editor/Viewport.cpp [hunk 2]
```diff
@@ -581,63 +590,7 @@ void QtViewport::keyReleaseEvent(QKeyEvent* event)
     OnKeyUp(nativeKey, 1, event->nativeModifiers());
 }
 
-//////////////////////////////////////////////////////////////////////////
-void QtViewport::OnLButtonDown(Qt::KeyboardModifiers modifiers, const QPoint& point)
```
Comment (NicholasVanSickle) [inline Code/Editor/Viewport.cpp:585]: ❤️

**Review by NicholasVanSickle — APPROVED** [review]
Nice one!

**Review by mbalfour-amzn — APPROVED** [review]

File: Code/Editor/2DViewport.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Editor/2DViewport.cpp:236]: Haha yes it's on my list 🙂 I'll see if I can get to it next week 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportMessages.cpp [hunk 3]
```diff
@@ -62,4 +63,34 @@ namespace AzToolsFramework
...
+
+    AZ::Vector3 CalculateWorldPosition(
+        AzFramework::ViewportId viewportId, const AzFramework::ScreenPoint& screenPoint, const float defaultDistance)
+    {
+        using AzToolsFramework::ViewportInteraction::ViewportInteractionRequestBus;
+        AzToolsFramework::ViewportInteraction::ProjectedViewportRay viewportRay{};
+        ViewportInteractionRequestBus::EventResult(
+            viewportRay, viewportId, &ViewportInteractionRequestBus::Events::ViewportScreenToWorldRay, screenPoint);
+
+        const float RayDistance = 1000.0f;
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportMessages.cpp:75]: Yeah good point, I can pull this out 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportMessages.cpp [hunk 4]
```diff
@@ -62,4 +63,34 @@ namespace AzToolsFramework
 
         return circleBoundWidth;
     }
+
+    AZ::Vector3 CalculateWorldPosition(
+        AzFramework::ViewportId viewportId, const AzFramework::ScreenPoint& screenPoint, const float defaultDistance)
```
[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportMessages.cpp:68]: Good call, I can update the name, will update tomorrow 👍 Thanks!

[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewportMessages.cpp:68]: I had a think about this for a while and settled on `FindClosestPickIntersection` which I think is most accurate (as we're taking a `ViewportId` and `ScreenPoint`). I can always update the name again in future (or factor out the code to multiple functions that we can use independently and compose).

**Pull request merged**